build_runner/lib/src/util/constants.dart

```diff
@@ -11,11 +11,18 @@ final String assetGraphPath = '$cacheDir/$scriptHash/asset_graph.json';
 /// Directories used for build tooling.
 ///
 /// Reading from these directories may cause non-hermetic builds.
-const toolDirs = const ['.dart_tool', 'build', 'packages', '.pub'];
+const toolDirs = const [
+  '.dart_tool',
+  'build',
+  'packages',
+  '.pub',
+  '.git',
```

Comment:
Should this list be configurable some how? Also I'm curious how reading from .idea and .git can cause non-hermetic builds.

Reply:
Ya these are folders we ignore, we might want to make it configurable in the future or possibly ignore all top level hidden dirs? I can file an issue for that.

Reply:
https://github.com/dart-lang/build/issues/440